identifier/identifier.go

```diff
@@ -122,6 +122,24 @@ func NewIP(ip netip.Addr) ACMEIdentifier {
 	}
 }
 
+// FromString converts a string to an ACMEIdentifier.
```

Comment:
I might like to see a comment on these two funcs with a warning that they shouldn't be used (in favour of NewDNS and NewIP) if you already know the type.